pandas/core/generic.py

```diff
@@ -11961,7 +11961,8 @@ def last_valid_index(self) -> Hashable:
 
 Returns
 -------
-{name1} or scalar\
+{name1} or scalar
+    Either {name1} or scalar based on ``axis`` selected.\
```

Comment:
Thanks for working on this @YashpalAhlawat. The idea here is to provide a description of what is being return. The first line is the type being returned, what you are adding should be a description, explaining conceptually what is being returned. Do you mind updating this to a description that is useful to users reading this PR to understand what the returned value means please.

Reply:
Sure @datapythonista  , I will.

In that case all different function will need different description. These docs are getting generated from common string which is getting formatted.

How to go ahead in that case.

Approach 1: I can add additional parameter for return desc , which will have return description. I can write the return description 

` docstr = base_doc.format(
        desc=desc,
        name=name,
        name1=name1,
        name2=name2,
        axis_descr=axis_descr,
        see_also=see_also,
        examples=examples,
        return_desc = return_desc,
        **kwargs,
    ) `
    
Approach 2:
    what I feel is all desc has first line stating what it is returning, We can use same first line. Instead of creating new variable but it will have the assumption first line of description for methods (min, max, mean, median, skew, kurt).
    which is true today but may not be in future
    e.g.:
    desc for max is:
    ` (
            "Return the maximum of the values over the requested axis.\n\n"
            "If you want the *index* of the maximum, use ``idxmax``. This is "
            "the equivalent of the ``numpy.ndarray`` method ``argmax``."
        ) `
       Similarly it follows for all.
       
What is your take on it?
       
**Also, isn't we are repeating same thing again and again in description of function and in the description of return .**

Reply:
We should have the functiom name as a parameter of the docstring already, as well as the object. Having something like "The result of applying the min function to the DataFrame elements, possibly just for rows or columns based on the axis parameter" seems reasonable and simple to implement.